lib/babel/generation/buffer.js

```diff
@@ -81,7 +82,7 @@ Buffer.prototype.newline = function (i, removeLast) {
 
   if (isNumber(i)) {
     if (this.endsWith("{\n")) i--;
-    if (this.endsWith(util.repeat(i, "\n"))) return;
+    if (this.endsWith(repeating("\n", i > 0 ? i : 0))) return;
```

Comment:
Added this conditional as `repeating` is strict and only allow positive numbers.